enterprise/frontend/src/embedding-sdk/store/auth/auth.ts

```diff
@@ -148,7 +129,10 @@ const getRefreshToken = async (
   );
   const { method, url: responseUrl, hash } = urlResponseJson || {};
   if (method === "saml") {
-    return await openSamlLoginPopup(responseUrl);
+    const token = await openSamlLoginPopup(responseUrl);
+    samlTokenStorage.set(token);
```

Comment:
I removed `samlTokenStorage.set(token)` from `openSamlLoginPopup` because that would set the SAML tokens in the parent, as it is being called in embed.js. So we do it in the SDK explicitly here.

Reply:
This is change a part of the iframe authentication?

I console long in this block and it isn't called in iframe embedding. What I read from this is that, this is only for the SDK (not the new iframe embedding).

And is there any problem with setting SAML token in the parent of the iframe? I don't get why we can't leave it there. The SDK also set the token in parent's localStorage (well since there's no iframe)

Reply:
Yes, this block is for SDK only. The `samlTokenStorage.set(token)` were part of `openSamlLoginPopup` which is used for the SDK, but I removed it so we can save the token outside of the `openSamlLoginPopup` function and choose where to store it.

> Is there any problem with setting SAML token in the parent of the iframe?

If the customer's website gets hacked, an attacker can't steal the session token if it's stored on the iframe as they can't execute cross-origin scripts on the iframe like `localStorage.getItem("SESSION_TOKEN")` on our iframe. If it's stored in the parent then it's easier for the tokens to get stolen.